File: BackEnd/backend/src/main/java/com/fitapp/backend/application/dto/RoutineSetParameter/request/UpdateSetParameterRequest.java
package com.fitapp.backend.application.dto.RoutineSetParameter.request;

import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UpdateSetParameterRequest {

    @JsonProperty("id")
    private Long id;

    @JsonProperty("parameterId")
     private Long parameterId;

     @JsonProperty("numericValue")
     private Double numericValue;

     @JsonProperty("durationValue")
     private Long durationValue;

     @JsonProperty("integerValue")
     private Integer integerValue;

     @JsonProperty("repetitions")
     private Integer repetitions;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }
}

File: FrontEnd/app/src/main/java/com/fitapp/appfit/feature/routine/model/setparameter/request/UpdateSetParameterRequest.kt
package com.fitapp.appfit.feature.routine.model.setparameter.request

import com.google.gson.annotations.SerializedName

data class UpdateSetParameterRequest(
    @SerializedName("id") val id: Long? = null,
    @SerializedName("parameterId") val parameterId: Long,
    @SerializedName("repetitions") val repetitions: Int? = null,
    @SerializedName("numericValue") val numericValue: Double? = null,
    @SerializedName("durationValue") val durationValue: Long? = null,
    @SerializedName("integerValue") val integerValue: Int? = null,
)